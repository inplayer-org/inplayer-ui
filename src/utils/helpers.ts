import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
