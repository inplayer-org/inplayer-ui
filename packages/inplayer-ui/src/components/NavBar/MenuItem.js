// @flow
import React, { type Node } from 'react';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';
import { transparentize } from 'polished';
import { uiColors, fontSizes } from 'utils';
import colors from 'config/colors';

const MenuItemContainer = styled.div`
  display: inline-flex;
  align-items: center;
  width: 100%;
  height: 50px;
  color: ${transparentize(0.4, colors.white)};
  transition: color 0.3s ease;
  overflow: hidden;
  white-space: nowrap;
  font-size: ${fontSizes('small')};
  ${ifProp(
    'active',
    css`
      background: ${uiColors('primary.dark')};
      border-right: ${uiColors('primary.main')} 3px solid;
      color: ${colors.white};
    `
  )};

  &:hover {
    color: ${colors.white};
  }
`;

const MenuItemIcon = styled.i`
  margin: 0 1rem;
  width: 20px;
  min-width: 20px;
`;

type MenuItemProps = {
  active: boolean,
  icon: ?string | ?Node,
  onClick: ?(e: SyntheticEvent<HTMLDivElement>) => any,
  children: ?Node,
};

const MenuItem = ({ active, icon, onClick, children }: MenuItemProps) => {
  const onItemClick = e => {
    e.stopPropagation();
    if (onClick && typeof onClick === 'function') {
      onClick(e);
    }
  };
  return (
    <MenuItemContainer active={active} onClick={onItemClick}>
      {typeof icon === 'string' ? (
        <MenuItemIcon className={icon} />
      ) : (
        <MenuItemIcon>{icon}</MenuItemIcon>
      )}

      {children}
    </MenuItemContainer>
  );
};

export default MenuItem;
