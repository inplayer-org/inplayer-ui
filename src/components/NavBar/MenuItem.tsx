// @flow
import React, { ReactNode, ChangeEvent } from 'react';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';
import { transparentize } from 'polished';
import { fontSizes } from 'utils';
import colors from 'theme/colors';

const MenuItemContainer = styled.div<{ onClick: (e: ChangeEvent<HTMLElement>) => any }>`
  display: inline-flex;
  align-items: center;
  width: 100%;
  height: 50px;
  color: ${transparentize(0.4, colors.white)};
  transition: color 0.3s ease;
  overflow: hidden;
  white-space: nowrap;
  font-size: ${fontSizes('small')};
  ${ifProp(
    'active',
    css`
      background: ${colors.navy};
      border-right: ${colors.skyBlue} 3px solid;
      color: ${colors.white};
    `
  )};

  &:hover {
    color: ${colors.white};
  }
`;

const MenuItemIcon = styled.i`
  margin: 0 1rem;
  width: 20px;
  min-width: 20px;
`;

interface MenuItemProps {
  active: boolean;
  icon?: string | ReactNode;
  onClick: ((e: React.MouseEvent<HTMLElement, MouseEvent>) => void) &
    ((e: React.ChangeEvent<HTMLElement>) => any);
  children?: ReactNode;
}

const MenuItem = ({ active, icon, onClick, children }: MenuItemProps) => {
  const onItemClick = (e: ChangeEvent<HTMLElement>) => {
    e.stopPropagation();
    if (onClick && typeof onClick === 'function') {
      onClick(e);
    }
  };
  return (
    <MenuItemContainer active={active} onClick={onItemClick}>
      {typeof icon === 'string' ? (
        <MenuItemIcon className={icon} />
      ) : (
        <MenuItemIcon>{icon}</MenuItemIcon>
      )}

      {children}
    </MenuItemContainer>
  );
};

export default MenuItem;
