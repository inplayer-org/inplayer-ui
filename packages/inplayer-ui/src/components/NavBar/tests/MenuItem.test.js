import { generateSnapshots } from 'packages/inplayer-ui/src/components/NavBar/tests/helpers';
import MenuItem from '../MenuItem';

generateSnapshots(MenuItem);
