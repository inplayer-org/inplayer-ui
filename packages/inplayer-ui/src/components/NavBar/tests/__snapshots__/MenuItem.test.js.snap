// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 50px;
  color: rgba(255,255,255,0.6);
  -webkit-transition: color 0.3s ease;
  transition: color 0.3s ease;
  overflow: hidden;
  white-space: nowrap;
  font-size: 0.875rem;
}

.c0:hover {
  color: #fff;
}

.c1 {
  margin: 0 1rem;
  width: 20px;
  min-width: 20px;
}

<div
  className="c0"
  onClick={[Function]}
>
  <i
    className="c1"
  />
</div>
`;
